src/graph/executor/algo/ShortestPathBase.h

```diff
@@ -86,6 +80,7 @@ class ShortestPathBase {
   std::unordered_map<std::string, std::string>* stats_{nullptr};
   size_t maxStep_;
   bool singleShortest_{true};
+  folly::SpinLock statsLock_;
```

Comment:
How about other memebers?

Reply:
other members will not be written by multiple threads at the same time